### PR TITLE
Use a TCP address for core agent communication

### LINKED ISSUE #184: Default to TCP socket for core agent communication
Too many installs are having issues with file permissions for the core agent unix socket. The core agent has the ability to listen on a TCP socket. Change the agent code to default to using only the TCP socket, but make sure we can still use the unix socket option too.

* Change the default `core_agent_socket_path` to `tcp://127.0.0.1:6590`.
* Any socket path that begins with tcp:// connect via a TCP socket instead of the current Unix Domain Socket
* Start the core agent with the `--tcp <bind>` flag to match.
* Ensure the socket behaves the same as with unix socket - e.g. reconnecting after unexpected disconnect, etc.

The argument to start the core agent with a TCP socket:

```
--tcp <tcp-bind>               TCP Socket to bind to, in form of ip:port (0.0.0.0:8383 for instance). If unset, no TCP socket is created.
```

See-Also [Matching Python Agent Issue](https://github.com/scoutapp/scout_apm_python/issues/553)

### PR DESCRIPTION
**Pull request by asgrim**
Fixes #184

- Added ConnectionAddress value object for TCP/socket path support
- Use ConnectionAddress to determine whether to use TCP or UNIX sockets
- Set default core_agent_socket_path configuration to TCP address 127.0.0.1:6590

### CODE REVIEW

**Comment by asgrim** [comment]
Updated to use https://github.com/scoutapp/core-agent/releases/tag/v1.3.0

File: src/Connector/ConnectionAddress.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    public function tcpBindAddress() : string
+    {
+        return explode(':', $this->tcpBindAddressPort(), 2)[0];
+    }
+
+    public function tcpBindPort() : int
+    {
+        return (int) explode(':', $this->tcpBindAddressPort(), 2)[1];
```
Comment (cschneid) [inline src/Connector/ConnectionAddress.php:69]: What happens if the user forgets to add a port, so tcpBindAddressPort is just `'127.0.0.1'`, and the `[1]` here is not a real index. Exception? And then is it caught politely?

**Review by cschneid — APPROVED** [review]
Just the one exception case question, otherwise all good!

File: src/Connector/ConnectionAddress.php (hunk 1 above)
Comment (asgrim) [reply src/Connector/ConnectionAddress.php:69]: I'll add testing around this to verify :+1:

**Pull request merged**